src/views/SpaceSettings.vue

```diff
@@ -287,6 +297,48 @@ watch(
   }
 );
 
+function handleEnsButtonClick() {
+  if (isOwner.value || isAdmin.value) {
+    window.open(`https://app.ens.domains/name/${props.spaceId}`, '_blank');
+    return;
+  }
+
+  handleSetRecord();
+}
+
+async function handleSetRecord() {
+  settingENSRecord.value = true;
+  try {
+    const ensPublicResolverAddress = networks[networkKey.value].ensResolver;
+    if (!ensPublicResolverAddress) {
+      throw new Error('No ENS resolver address for this network');
+    }
+    const ensname = props.spaceId;
+    const node = namehash.hash(ensname);
+    const tx = await sendTransaction(
+      auth.web3,
+      ensPublicResolverAddress,
+      abi,
+      'setText',
+      [node, 'snapshot', textRecord.value]
+    );
+    const receipt = await tx.wait(2);
+    console.log('Receipt', receipt);
+    const uri = await getSpaceUri(
+      props.spaceId,
+      import.meta.env.VITE_DEFAULT_NETWORK
+    );
```

Comment:
> It is possible that something might not work since we assume to get the spaceUri on mainnet.

Actually we are not, thanks to this env var. I have everything set up to Rinkeby and it works fine. :)